### PR TITLE
Package tuplet: A Lightweight Tuple Library for Modern C++

### PR DESCRIPTION
**Pull request by BlueSolei**
Library: **tuplet/1.2.2**

std::tuple is a vocabulary type, a fundamental type that is used everywhere.
this library drastically enhances tuple's performance and other very important characteristics - triviality (default ctor, copy, move, assignment). these characteristics are fundamental for low-level architectures. it simplifies the generated code, and enables copy between address spaces, using of registers, and other optimizations.
since the standard mandates the preservation of ABI of std::tuple, and other backward compatibility demands, the implementation of std::tuple is sub-par.

[Here is the motivation behind `tuplet`, written by its author](https://github.com/codeinred/tuplet#Motivation)

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/tuplet/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+from conans.errors import ConanInvalidConfiguration
+
+
+class TupletConan(ConanFile):
+    name = "tuplet"
+    license = "BSL-1.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/codeinred/tuplet"
+    description = "A fast, simple tuple implementation that implements tuple as an aggregate"
+    topics = ("tuple", "trivially-copyable", "modern-c++")
```
Comment (uilianries) [inline recipes/tuplet/all/conanfile.py:12]:
```suggestion
    topics = ("tuple", "trivially-copyable", "modern-cpp")
```
Avoid spaces and `+` . Search engines are trick and can understand it as `modern-c` only

File: recipes/tuplet/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                f"{self.name} {self.version} requires C++20, which your compiler ({compiler}-{version}) does not support")
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package(self):
+        include_folder = os.path.join(self._source_subfolder, "include")
+        self.copy(pattern="*.hpp", dst="include", src=include_folder)
+        self.copy(pattern="LICENSE", dst="licenses",
+                  src=self._source_subfolder)
```
Comment (uilianries) [inline recipes/tuplet/all/conanfile.py:70]:
```suggestion
        tools.files.copy(self, pattern="*.hpp", dst=os.path.join(self.package_folder, "include"), src=include_folder)
        tools.files.copy(self, pattern="LICENSE", dst=os.path.joins(self.package_folder, "licenses"),
                  src=self._source_subfolder)
```
We are moving to Conan 2.0 and our linter will complain about old tools method.

File: recipes/tuplet/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import os
+from conan import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
```
Comment (uilianries) [inline recipes/tuplet/all/conanfile.py:3]:
```suggestion
from conan.errors import ConanInvalidConfiguration
```

**Review by uilianries — APPROVED** [review]
LGTM

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**